src/firefly/js/ui/UploadTableChooser.js

```diff
@@ -81,14 +81,14 @@ function uploadSubmit(request, setUploadInfo)  {
     const columns= report.fileFormat===Format.FITS ?
         getFitsColumnInfo(data) :
         data.map(([name,type,u,d]) => ({name, type, units: u?u:'', description: d?d:'', use:true}));
-    const columnsSelected = getSelectedColumns(columns);
+    const columnsSelected = getSelectedColumns(columns,defaultColsEnabled);
```

Comment:
Assuming you take the suggested function approach and called parameter `defaultColsSelector`, then this line can be...
```
const columnsSelected = defaultColsSelector?.(columns) || applyDefColumnSelection(columns);
```